arroyo-worker/src/operators/windows.rs

```diff
@@ -324,15 +324,15 @@ impl<K: Key, T: Data, OutT: Data> SessionWindowFunc<K, T, OutT> {
                 delete_behavior: TableDeleteBehavior::NoReadsBeforeWatermark as i32,
                 write_behavior: TableWriteBehavior::NoWritesBeforeWatermark as i32,
                 // we always write the largest end in the list of windows
-                retention_micros: 0 as u64,
+                retention_micros: MAX_SESSION_SIZE.as_micros() as u64,
             },
             TableDescriptor {
                 name: "s".to_string(),
                 description: "sessions".to_string(),
                 table_type: TableType::TimeKeyMap as i32,
                 delete_behavior: TableDeleteBehavior::NoReadsBeforeWatermark as i32,
                 write_behavior: TableWriteBehavior::NoWritesBeforeWatermark as i32,
-                retention_micros: MAX_SESSION_SIZE.as_micros() as u64,
+                retention_micros: 0 as u64,
```

Comment:
Why not `MAX_SESSION_SIZE`?

Reply:
The timestamp is the end time of the final session, so we don't need to keep any data before the watermark, as it'll have already been processed.